renderdoc/driver/gl/gl_debug.cpp

```diff
@@ -1165,7 +1185,7 @@ uint32_t GLReplay::PickVertex(uint32_t eventID, const MeshDisplay &cfg, uint32_t
       (GLsizeiptr)(cfg.position.idxoffs + cfg.position.idxByteWidth * cfg.position.numVerts));
   gl.glBindBufferBase(eGL_SHADER_STORAGE_BUFFER, 3, DebugData.pickResultBuf);
 
-  gl.glDispatchCompute(GLuint(cfg.position.numVerts / 128 + 1), 1, 1);
+  gl.glDispatchCompute(GLuint((cfg.position.numVerts / 3) / 128 + 1), 1, 1); //launch one thread per triangle
```

Comment:
This is the only non-nitpick issue I can find. This looks to me like it will only work for triangle list topologies - right? You mention lines and point topologies which would also need some kind of fallback, but this method would also have to handle triangle strips (common), triangle adjacency lists/strips (less common) and triangle fans (pretty rare, not sure if anyone uses them these days). The tessellation patch list topologies I tend to just display as points so that would use the same fallback as pointlists.

I don't know what the best solution is, whether to make the compute shader more complex to be aware of how to convert primitive ID to triangle indices, or whether to do an expansion step first to an intermediate buffer then run the existing CS. I suspect the former should be feasible as the topologies aren't wildly complicated, but I don't know as I haven't tried it :smile:.